waiter/src/waiter/request_log.clj

```diff
@@ -0,0 +1,81 @@
+;;
+;;       Copyright (c) 2018 Two Sigma Investments, LP.
+;;       All Rights Reserved
+;;
+;;       THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF
+;;       Two Sigma Investments, LP.
+;;
+;;       The copyright notice above does not evidence any
+;;       actual or intended publication of such source code.
+;;
+(ns waiter.request-log
+  (:require [clj-time.core :as t]
+            [clojure.core.async :as async]
+            [clojure.data.json :as json]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [metrics.timers :as timers]
+            [waiter.async-utils :as au]
+            [waiter.metrics :as metrics]
+            [waiter.ring-utils :as ru]
+            [waiter.utils :as utils]))
+
+(defn log
+  "Log log-data as JSON"
+  [log-data]
+  (log/log "RequestLog" :info nil (json/write-str log-data :escape-slash false)))
+
+(defn request->context
+  "Convert a request into a context suitable for logging."
+  [{:keys [headers query-string remote-addr request-id request-method request-time uri] :as request}]
+  (let [{:strs [host x-cid x-forwarded-for]} headers]
+    (cond-> {:cid x-cid
+             :host host
+             :path uri
+             :remote-addr (or x-forwarded-for remote-addr)
+             :request-id request-id
+             :scheme (let [scheme (utils/request->scheme request)]
```

Comment:
Can `scheme` be `nil`? Else we can call `name` on it directly. Else: `(some-> (utils/request->scheme request) name)`

Reply:
I don't believe so.  If it is, then `name` is only called if it's a keyword.

Reply:
Scheme can be a string if it's coming from haproxy via x-forwarded-proto

Reply:
```
(name "some-string") => "some-string"
(name :some-keyword) => "some-keyword"
```

Reply:
Do you want to change or should I go ahead and merge?